### ISSUE TITLE
'TypeError: Path must be a string' related to autoprefixer

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#2775]
I'm tracing a error that seems to only occur when stylelint is run via a npm script (works standalone):

```js
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:1167:7)
    at eachParent (node_modules/browserslist/index.js:65:18)
    at getStat (node_modules/browserslist/index.js:79:12)
    at browserslist (node_modules/browserslist/index.js:188:15)
    at Browsers.parse (node_modules/autoprefixer/lib/browsers.js:61:16)
    at new Browsers (node_modules/autoprefixer/lib/browsers.js:52:30)
    at Object.<anonymous> (node_modules/stylelint/lib/utils/isAutoprefixable.js:19:3)
```

When I try the reduced code from [the throwing line](https://github.com/stylelint/stylelint/blob/1631199ff3a35a155510ea936fbfda1f8414d28c/lib/utils/isAutoprefixable.js#L19), I can reproduce the throw in the node REPL:

```js
$ node
> const autoprefixer = require("autoprefixer");
> const Browsers = require("autoprefixer/lib/browsers");
> new Browsers(autoprefixer.data.browsers, []);

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.resolve (path.js:1167:7)
    at eachParent (node_modules/browserslist/index.js:65:18)
    at getStat (node_modules/browserslist/index.js:79:12)
    at browserslist (node_modules/browserslist/index.js:188:15)
    at Browsers.parse (node_modules/autoprefixer/lib/browsers.js:61:16)
    at new Browsers (node_modules/autoprefixer/lib/browsers.js:52:30)
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
    at REPLServer.defaultEval (repl.js:239:29)
```

```console
$ npm ls browserslist
└─┬ stylelint@8.0.0
  └─┬ autoprefixer@7.1.2
    └── browserslist@2.2.1
```

Anyone able to reproduce?

**Comment by xPaw** [comment]
I am getting the same error on my CIs, but not locally.

I checked the local version and I have `browserslist@2.1.5` which does not have this problem.

**Comment by bjankord** [comment]
I'm seeing this error as well in one of our projects. Details below.



> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug or a config question

> Which rule, if any, is this issue related to?

isAutoprefixable

> What CSS is needed to reproduce this issue?

```css
.foo {
  color: #f00;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-sass-guidelines",
  "plugins": [
     "stylelint-suitcss"
  ],
  "rules": {
    "max-nesting-depth": 3,
    "scss/at-mixin-pattern": "^(terra-)[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
    "suitcss/custom-property-no-outside-root": true,
    "custom-property-pattern": [
      "terra-[a-z]+([a-z0-9-]+[a-z0-9]+)?$",
      {
        "message": "Custom property names should be written in lowercase with hyphens"
      }
    ],
    "selector-class-pattern": [
      "^terra-(?:[A-Z][a-zA-Z0-9]+)*(?:-[a-z0-9][a-zA-Z0-9]+)*?(?:--[a-z0-9][a-zA-Z0-9-]+)*?$|^(?:terra-)?(?:u|is|has|js)(?:-[a-z0-9][a-zA-Z0-9-]*)+$",
      {
        "message": "Class names should match the SUIT CSS naming convention"
      }
    ]
  }
}
```

> Which version of stylelint are you using?

e.g. `8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

We run it from an npm-script inside a monorepo

`"lint:scss": "$(cd ..; npm bin)/stylelint src/**/*.scss",`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

We use SCSS, but I don't think this issue is related to SCSS. Not sure though.

> What did you expect to happen?

No errors to be thrown.

> What actually happened (e.g. what warnings or errors you are getting)?

The following TypeError is thrown:

```shell
path.js:7
    throw new TypeError('Path must be a string. Received ' + inspect(path));
    ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.resolve (path.js:1146:7)
    at eachParent (/Users/bj031910/repos/terra-core/node_modules/browserslist/index.js:65:18)
    at getStat (/Users/bj031910/repos/terra-core/node_modules/browserslist/index.js:79:12)
    at browserslist (/Users/bj031910/repos/terra-core/node_modules/browserslist/index.js:188:15)
    at Browsers.parse (/Users/bj031910/repos/terra-core/node_modules/autoprefixer/lib/browsers.js:61:16)
    at new Browsers (/Users/bj031910/repos/terra-core/node_modules/autoprefixer/lib/browsers.js:52:30)
    at Object.<anonymous> (/Users/bj031910/repos/terra-core/node_modules/stylelint/lib/utils/isAutoprefixable.js:19:3)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
```

**Comment by Ladsgroup** [comment]
I'm getting this in local machine and travis too: https://travis-ci.org/wiki-ai/wikilabels/builds/256334402?utm_source=github_status&utm_medium=notification

**Comment by silverwind** [comment]
I think this maybe because of this line:

https://github.com/ai/browserslist/commit/9d3e5de995a5f560452c6f9457f41e25b5634924#diff-168726dbe96b3ce427e7fedce31bb0bcR81

`getStat` previously always returned a string from that loop, can now return `undefined`. cc: @ai

**Comment by silverwind** [comment]
Closing, see https://github.com/ai/browserslist/issues/158

**Comment by bjankord** [comment]
For those tracking on this issue, there is a PR which fixes the issue in browserslist here: https://github.com/ai/browserslist/pull/161

**Comment by ai** [comment]
Fixed in Browserslist 2.2.2

**Comment by ntwb** [comment]
Thanks everyone for helping document the issue :)

Thanks for the quick turnaround on shipping a fix @ai :)
